crates/circuit/src/circuit_data.rs

```diff
@@ -341,6 +341,40 @@ impl CircuitData {
         Ok(self_)
     }
 
+    /// Put ``self`` into the canonical physical form, with the given number of qubits.
+    ///
+    /// This acts in place, and does not need to traverse the circuit.  It is intended for use when
+    /// the circuit is known to already represent a physical circuit, and we just need to assert
+    /// that it is canonical physical form.
+    ///
+    /// This erases any information about virtual qubits in the :class:`CircuitData`; if using this
```

Comment:
We don't publish the docs for `CircuitData` it's an internal class so this ref doesn't point to anything. We do document `QuantumCircuit.data` but just as a list-like object of `CircuitInstructions`. Granted this is all internal docs anyway so we're not going to build this page directly so it doesn't matter.

Reply:
I don't really know why I put in Sphinx markup that won't do anything, but I like to generally document stuff that needs some care, even if it's only internal and will only be read in a code comment.  Though in Rust, I usually write markdown...

Reply:
Ohhh, I know why - the method is straight-up copy-pasted from `DAGCircuit`, where it _was_ public documentation.